### ISSUE TITLE
[pro] PunditPolicy::Scope#resolve is not called with :resolver option

### ISSUE THREAD

**Issue by QWYNG** [rmosolgo/graphql-ruby#3989]
**Describe the bug**
In `GraphQL::Schema::Object#field` with :resolver option to return connection type,
PunditPolicy::Scope#resolve is called in 2.0.0 or before but not in 2.0.1 or newer.

**Versions**

`graphql` version: 2.0.2 & 2.0.1
` activerecord` version : 7.0.2.3
`pundit` version: 2.2.0

**Steps to reproduce**

```ruby
require 'bundler/inline'

gemfile(true) do
  source 'https://rubygems.org'
  gem 'activerecord'
  gem 'graphql', '2.0.0'
  gem 'graphql-pro', '1.21.6', source: 'https://gems.graphql.pro/'
  gem 'pundit', '2.2.0'
  gem 'sqlite3'
end

require 'active_record'

ActiveRecord::Base.establish_connection(
  adapter: 'sqlite3',
  database: ':memory:'
)

ActiveRecord::Schema.define do
  create_table :items do |table|
  end
end

class Item < ActiveRecord::Base
end

2.times { |_n| Item.create }

class ItemPolicy
  def initialize(_user, _object); end

  class Scope
    def initialize(_account, scope)
      @scope = scope
    end

    def resolve
      puts 'scoping'
      @scope.all
    end
  end
end

class BaseObject < GraphQL::Schema::Object
  include GraphQL::Pro::PunditIntegration::ObjectIntegration
end

class ItemType < BaseObject
  pundit_policy_class ItemPolicy
  pundit_role nil

  field :id, Integer, null: true
end

class ItemConnectionType < GraphQL::Types::Relay::BaseConnection
  include GraphQL::Pro::PunditIntegration::ObjectIntegration
  pundit_role nil
  pundit_policy_class ItemPolicy
  edge_type(ItemType.edge_type)
end

class ItemResolver < GraphQL::Schema::Resolver
  type ItemConnectionType, null: false

  def resolve
    Item.all
  end
end

class QueryType < BaseObject
  pundit_role nil

  field :items, resolver: ::ItemResolver
end

class TestSchema < GraphQL::Schema
  query(QueryType)
end

query = <<-GRAPHQL
  query {
    items {
      nodes {
        id
      }
    }
  }
GRAPHQL

TestSchema.execute(query, variables: {}, context: {})
```
**Expected behavior**

ItemPolicy::Scope#resolve will call with all version

```
-- create_table(:items)
   -> 0.0037s
scoping
scoping
```

**Actual behavior**

with ` gem 'graphql', '2.0.0`

```
-- create_table(:items)
   -> 0.0037s
scoping
scoping
```

with ` gem 'graphql', '2.0.1'`

```
-- create_table(:items)
   -> 0.0041s
```

with ` gem 'graphql', '2.0.2'`

```
-- create_table(:items)
   -> 0.0042s
```

**Additional context**

Without `:resolver` option, ItemPolicy::Scope#resolve will be called in any version.

```ruby
require 'bundler/inline'

gemfile(true) do
  source 'https://rubygems.org'
  gem 'activerecord'
  gem 'graphql', '2.0.0'
  gem 'graphql-pro', '1.21.6', source: 'https://gems.graphql.pro/'
  gem 'pundit', '2.2.0'
  gem 'sqlite3'
end

require 'active_record'

ActiveRecord::Base.establish_connection(
  adapter: 'sqlite3',
  database: ':memory:'
)

ActiveRecord::Schema.define do
  create_table :items do |table|
  end
end

class Item < ActiveRecord::Base
end

2.times { |_n| Item.create }

class ItemPolicy
  def initialize(_user, _object); end

  class Scope
    def initialize(_account, scope)
      @scope = scope
    end

    def resolve
      puts 'scoping'
      @scope.all
    end
  end
end

class BaseObject < GraphQL::Schema::Object
  include GraphQL::Pro::PunditIntegration::ObjectIntegration
end

class ItemType < BaseObject
  pundit_policy_class ItemPolicy
  pundit_role nil

  field :id, Integer, null: true
end

class ItemConnectionType < GraphQL::Types::Relay::BaseConnection
  include GraphQL::Pro::PunditIntegration::ObjectIntegration
  pundit_role nil
  pundit_policy_class ItemPolicy
  edge_type(ItemType.edge_type)
end

class QueryType < BaseObject
  pundit_role nil

  field :items, ItemConnectionType, null: false

  def items
    Item.all
  end
end

class TestSchema < GraphQL::Schema
  query(QueryType)
end

query = <<-GRAPHQL
  query {
    items {
      nodes {
        id
      }
    }
  }
GRAPHQL

TestSchema.execute(query, variables: {}, context: {})
```

output 

with ` gem 'graphql', '2.0.0`

```
-- create_table(:items)
   -> 0.0037s
scoping
scoping
```

with ` gem 'graphql', '2.0.1'`

```
-- create_table(:items)
   -> 0.0041s
scoping
scoping
```

with ` gem 'graphql', '2.0.2'`

```
-- create_table(:items)
   -> 0.0042s
scoping
scoping
```

**Comment by rmosolgo** [comment]
Thank you for the detailed report! I _bet_ it's a bug from this change: https://github.com/rmosolgo/graphql-ruby/pull/3916/files

I'll take a look and follow up here.

**Comment by rmosolgo** [comment]
Sorry for the trouble! It should be fixed on master and it'll be released in 2.0.3 soon. Please let me know if you find any more trouble after that.

**Comment by QWYNG** [comment]
@rmosolgo thank you so much！
